lib/matplotlib/backend_bases.py

```diff
@@ -3076,24 +3074,20 @@ def pan(self, *args):
 
     def press_pan(self, event):
         """Callback for mouse button press in pan/zoom mode."""
-        if event.button in [1, 3]:
-            self._button_pressed = event.button
-        else:
-            self._button_pressed = None
+        if event.button not in [1, 3] or event.x is None or event.y is None:
+            return
+        axes = [a for a in self.canvas.figure.get_axes()
+                if a.in_axes(event) and a.get_navigate() and a.can_zoom()]
+        if not axes:
             return
         if self._nav_stack() is None:
-            # set the home button to this view
-            self.push_current()
-        x, y = event.x, event.y
-        self._xypress = []
-        for i, a in enumerate(self.canvas.figure.get_axes()):
-            if (x is not None and y is not None and a.in_axes(event) and
-                    a.get_navigate() and a.can_pan()):
-                a.start_pan(x, y, event.button)
-                self._xypress.append((a, i))
-                self.canvas.mpl_disconnect(self._id_drag)
-                self._id_drag = self.canvas.mpl_connect(
-                    'motion_notify_event', self.drag_pan)
+            self.push_current()  # set the home button to this view
+        for ax in axes:
+            ax.start_pan(event.x, event.y, event.button)
+        self.canvas.mpl_disconnect(self._id_drag)
+        id_drag = self.canvas.mpl_connect("motion_notify_event", self.drag_pan)
+        self._pan_info = {
```

Comment:
Could the info objects be namedtuples? That'd be more clear than an arbitrary dict.

Reply:
The Advantage of a namedtuple is that it's immutable:

- The information is complete. It will always have exactly the three given attributes.
- Nobody will change that information later.

With dicts, you'd have to check all usages of `_pan_info` to be sure to know how it works.

I'd be fine with `ZoomInfo(button=event.button, axes=axes, cid=id_drag)`.

I'm still in favoe of the namedtuple, but won't fight over this.

Reply:
I'll leave it to reviewer 2 to pick a side :)

Reply:
I don't have a high preference either way, but a namedtuple is nice as being self-documenting, somewhat.